py3/game.py

```diff
@@ -1151,39 +1167,44 @@ def lotto():
         master.bell()
         lottoafford = Label(master, text="%s" % cannotafford, width=35)
         lottoafford.grid(row=8, column=0, sticky=W)
-        master.after(500, lambda: eval('''lottoafford.destroy()
-lottobutton = Button(master, width=35, text="Lotto ($%s)" % (lottoprice)')
-lottobutton.grid(row=11, column=0, sticky=W)'''))
+
+        def update_lotto_afford(price):
+            lottoafford.destroy()
+            lottobutton = Button(master, width=35, text="Lotto ($%s)" % (price))
+            lottobutton.grid(row=11, column=0, sticky=W)
+
+        master.after(500, lambda: update_lotto_afford(game_state.lottoprice))
     else:
         global lottoprice
-        money = game_state.money - game_state.lottoprice
+        lp = game_state.lottoprice
+        money = game_state.money - lp
         game_state.lottoprice *= uniform(1.1, 5.1)
         prob = random()
         if prob < 0.5:
-            money += (lottoprice / 2.0)
+            money += (lp / 2.0)
         elif prob < 0.7:
-            money += (lottoprice / 1.5)
+            money += (lp / 1.5)
         elif prob < 0.8:
-            money += lottoprice
+            money += lp
         elif prob < 0.8625:
-            money += lottoprice * 1.5
+            money += lp * 1.5
         elif prob < 0.8825:
-            money += lottoprice * 2.5
+            money += lp * 2.5
         elif prob < 0.895:
-            money += lottoprice * 4
+            money += lp * 4
         elif prob < 0.896:
-            money += lottoprice * 10
+            money += lp * 10
         elif prob < 0.8965:
-            money += lottoprice * 40
+            money += lp * 40
         elif prob < 0.8967:
-            money += lottoprice * 250
+            money += lp * 250
         elif prob < 0.896875:
-            money += lottoprice * 1000
+            money += lp * 1000
         elif prob < 0.897:
-            money += lottoprice * 20000
+            money += lp * 20000
         else:
             money /= 2.0
-        lottoprice.set('Lotto ($' + str(round(game_state.lottoprice, 1)) + ')')
+        lottoprice.set('Lotto ($' + str(round(lp, 1)) + ')')
```

Comment:
@qlstudio what do you mean?  stuff was pretty broken when i did a fresh clone...